### PR TITLE
escape few characters in url

### PR DESCRIPTION
**Pull request by madanagopalt**

### CODE REVIEW

File: examples/pxScene2d/src/pxScene.cpp [hunk 1]
```diff
@@ -112,14 +112,26 @@ class sceneWindow : public pxWindow, public pxIViewContainer
   {
     pxWindow::init(x,y,w,h);
 
+    // escape url begin
+    std::string escapedUrl;
+    std::string origUrl = url;
+    for (size_t index=0; index<origUrl.length(); index++)
```
Comment (andrewbenton) [inline examples/pxScene2d/src/pxScene.cpp]: Ideally, this should use the c++ iterator.

File: examples/pxScene2d/src/pxScene.cpp [hunk 2]
```diff
@@ -112,14 +112,26 @@ class sceneWindow : public pxWindow, public pxIViewContainer
...
+    // escape url end
     char buffer[1024];
-		std::string urlStr(url);
-		if (std::string::npos != urlStr.find("http")) {
-    sprintf(buffer,"shell.js?url=%s",rtUrlEncodeParameters(url).cString());
-		}
-		else {
-			sprintf(buffer, "shell.js?url=%s",url);
-		}
+    if (std::string::npos != escapedUrl.find("http")) {
```
Comment (andrewbenton) [inline examples/pxScene2d/src/pxScene.cpp]: Do we not want to encode parameters for websocket connections?

File: examples/pxScene2d/src/pxScene2d.cpp [hunk 3]
```diff
@@ -2942,9 +2956,9 @@ void pxScriptView::runScript()
 
     char buffer[1024];
 #ifdef RUNINMAIN
-    sprintf(buffer, "loadUrl(\"%s\");", url);
+    snprintf(buffer, sizeof(buffer), "loadUrl(\"%s\");", mUrl.cString());
 #else
-    sprintf(buffer, "loadUrl(\"%s\");", mUrl.cString());
+    snprintf(buffer, sizeof(buffer), "loadUrl(\"%s\");", mUrl.cString());
```
Comment (andrewbenton) [inline examples/pxScene2d/src/pxScene2d.cpp]: If these are the same in both paths of the define, they should probably be moved outside of the define section.

File: examples/pxScene2d/src/pxScene2d.cpp [hunk 4]
```diff
@@ -2921,6 +2920,21 @@ void pxScriptView::runScript()
   rtLogInfo(__FUNCTION__);
 #endif // ifndef RUNINMAIN
 
+// escape url begin
+  string escapedUrl;
+  string origUrl = url;
+  for (size_t index=0; index<origUrl.length(); index++)
```
Comment (andrewbenton) [inline examples/pxScene2d/src/pxScene2d.cpp]: C++ iterator pattern as above

File: examples/pxScene2d/src/pxScene.cpp (hunk 1 above)
Comment (madanagopalt) [inline examples/pxScene2d/src/pxScene.cpp]: addressed

File: examples/pxScene2d/src/pxScene.cpp [hunk 5]
```diff
@@ -112,14 +112,26 @@ class sceneWindow : public pxWindow, public pxIViewContainer
...
+    {
+      char currChar = origUrl.at(index);
+      if ((currChar == '"') || (currChar == '\\'))
+      {
+        escapedUrl.append(1, '\\');
+      }
+      escapedUrl.append(1, currChar);
+    }
+    // escape url end
     char buffer[1024];
```
Comment (madanagopalt) [inline examples/pxScene2d/src/pxScene.cpp]: addressed

File: examples/pxScene2d/src/pxScene.cpp [hunk 6]
```diff
@@ -112,14 +112,26 @@ class sceneWindow : public pxWindow, public pxIViewContainer
...
-		std::string urlStr(url);
-		if (std::string::npos != urlStr.find("http")) {
-    sprintf(buffer,"shell.js?url=%s",rtUrlEncodeParameters(url).cString());
-		}
-		else {
-			sprintf(buffer, "shell.js?url=%s",url);
-		}
+    if (std::string::npos != escapedUrl.find("http")) {
+      sprintf(buffer,"shell.js?url=%s",rtUrlEncodeParameters(escapedUrl.c_str()).cString());
```
Comment (madanagopalt) [inline examples/pxScene2d/src/pxScene.cpp]: addressed

File: examples/pxScene2d/src/pxScene.cpp [hunk 7]
```diff
@@ -112,14 +112,26 @@ class sceneWindow : public pxWindow, public pxIViewContainer
...
+    }
+    else {
+      sprintf(buffer, "shell.js?url=%s",escapedUrl.c_str());
```
Comment (madanagopalt) [inline examples/pxScene2d/src/pxScene.cpp]: addressed

File: examples/pxScene2d/src/pxScene2d.cpp (hunk 4 above)
Comment (madanagopalt) [inline examples/pxScene2d/src/pxScene2d.cpp]: addressed

File: examples/pxScene2d/src/pxScene2d.cpp (hunk 3 above)
Comment (madanagopalt) [inline examples/pxScene2d/src/pxScene2d.cpp]: addressed

**Comment by andrewbenton** [comment]
:+1:

**Pull request merged**